File: app/src/main/java/com/lcarino/bucketlist/ui/list/ListView.java
package com.lcarino.bucketlist.ui.list;

import com.lcarino.bucketlist.model.ui.BucketListItemViewModel;
import com.lcarino.bucketlist.mvp.MvpView;
import com.lcarino.bucketlist.ui.list.model.Inspiration;

import java.util.List;

/**
 * @author Luis Carino.
 */

public interface ListView extends MvpView {

    void showProgress();

    void hideProgress();

    void displayListItems(List<Inspiration> items);

    void clearInputField();

    void scrollToBottom();

    void displayList(List<BucketListItemViewModel> items);

}

File: app/src/main/java/com/lcarino/bucketlist/ui/list/ListView.kt
package com.lcarino.bucketlist.ui.list

import com.lcarino.bucketlist.model.ListEntry
import com.lcarino.bucketlist.model.ui.BucketListItemViewModel
import com.lcarino.bucketlist.mvp.MvpView
import io.realm.RealmResults

/**
 * Defines list view actions.
 * @author Luis Carino.
 */

interface ListView : MvpView {

    fun clearInputField()

    fun scrollToBottom()

    fun displayList(items: List<BucketListItemViewModel>)

    fun showArchivedITems(results: RealmResults<ListEntry>)

    fun displayAlertDialog()

}
